invenio_vocabularies/datastreams/readers.py

```diff
@@ -116,6 +117,32 @@ def read(self, item=None, *args, **kwargs):
         yield from self._iter(url=url, *args, **kwargs)
 
 
+class RORHTTPReader(BaseReader):
+    """ROR HTTP Reader returning an in-memory binary stream of the latest ROR data dump ZIP file."""
+
+    def _iter(self, fp, *args, **kwargs):
+        raise NotImplementedError(
+            "RORHTTPReader downloads one file and therefore does not iterate through items"
+        )
+
+    def read(self, item=None, *args, **kwargs):
+        """Reads the latest ROR data dump ZIP file from Zenodo and yields an in-memory binary stream of it."""
+        if item:
+            raise NotImplementedError(
+                "RORHTTPReader does not support being chained after another reader"
+            )
+
+        api_url = "https://zenodo.org/api/communities/ror-data/records?q=&sort=newest"
+        api_resp = requests.get(api_url)
+        api_resp.raise_for_status()
+
+        file_url = api_resp.json()["hits"]["hits"][0]["files"][0]["links"]["self"]
+        file_resp = requests.get(file_url)
+        file_resp.raise_for_status()
+
+        yield io.BytesIO(file_resp.content)
```

Comment:
Here the whole file is loaded in memory.
I tried directly returning `file_resp.raw`, but then converting to `ZipFile` fails due to the fact that `file_resp.raw` is not seekable.